### PR TITLE
[plugin.video.iranintl] v1.0.0

### PR DESCRIPTION
**Pull request by bbk79**
Iran International TV - Initial release

### Description
Initial release of Iran International TV add-on, a UK based Persian TV channel.
https://iranintl.com/

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.iranintl/default.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+__settings__ = xbmcaddon.Addon()
+home = __settings__.getAddonInfo('path')
+addon_handle = int(sys.argv[1])
+icon = xbmc.translatePath(os.path.join(home, 'icon.png'))
+li = xbmcgui.ListItem ('Live Stream')
+li.setArt({ 'thumb': icon })
+li.setInfo('video', {'plot':'Iran International TV Live Stream'})
+
+liveVideoUrl = "https://live.playstop.live/LS-63503-1/index.m3u8"
```
Comment (basrieter) [inline plugin.video.iranintl/default.py]: Could you provide us the site where you got this URL from? Just to check the source is available and free for the public?

Comment (bbk79) [reply plugin.video.iranintl/default.py]: Sure! https://iranintl.com/live

Comment (basrieter) [reply plugin.video.iranintl/default.py]: Seems like that one is pointing to https://live.livetvstream.co.uk/LS-63503-3/index.m3u8

Comment (bbk79) [reply plugin.video.iranintl/default.py]: You're right, that's the current behavior. I'll update the URL!

    function pickOrigin()
    {
        return "https://live.livetvstream.co.uk/LS-63503-3/index.m3u8";
        if(Math.random() < EXPERIMENT_ORIGIN_FRAC) {
            return "https://dev-live.livetvstream.co.uk/LS-63503-4/index.m3u8";
        }
        return "https://live.playstop.me/1816184091/index.m3u8";
    }

Comment (basrieter) [reply plugin.video.iranintl/default.py]: I would say you fetch https://iranintl.com/live and extract the URL from it. That way you will always be right?

File: plugin.video.iranintl/addon.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  <requires>
+    <import addon="xbmc.python" version="2.14.0"/>
+  </requires>
+  <extension point="xbmc.python.pluginsource"
+            library="default.py">
+        <provides>video</provides>
+  </extension>
+  <extension point="xbmc.addon.metadata">
+    <summary lang="en">Iran International TV</summary>
+    <description lang="en">Launched in May 2017, Iran International TV is a mix of rolling news, current affairs and documentaries, sport, lifestyle, health and technology, programmed to attract a wide audience, especially younger people. Iran International is a privately-owned UK entity.</description>
```
Comment (basrieter) [inline plugin.video.iranintl/addon.xml]: en_GB

File: plugin.video.iranintl/addon.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+       provider-name="babak">
+  <requires>
+    <import addon="xbmc.python" version="2.14.0"/>
+  </requires>
+  <extension point="xbmc.python.pluginsource"
+            library="default.py">
+        <provides>video</provides>
+  </extension>
+  <extension point="xbmc.addon.metadata">
+    <summary lang="en">Iran International TV</summary>
```
Comment (basrieter) [inline plugin.video.iranintl/addon.xml]: en_GB

File: plugin.video.iranintl/default.py (hunk 1 above)
Comment (bbk79) [reply plugin.video.iranintl/default.py]: Yes, I generally prefer that approach too :) Need to execute JS in this case though as parsing alone won't do. Is there a phantom.js module I can import? Some older documentation suggests that I might have to provide binaries for every platform?

File: plugin.video.iranintl/addon.xml (hunk 2 above)
Comment (bbk79) [reply plugin.video.iranintl/addon.xml]: Sure but I'm pretty sure @Rechi once got me to do the opposite on a different add-on?

**Comment by bbk79** [comment]
Travis tests seem to have failed because of a timeout, can you please trigger a re-run @basrieter?

File: plugin.video.iranintl/default.py (hunk 1 above)
Comment (basrieter) [reply plugin.video.iranintl/default.py]: You can just fetch the js url data and parse it using a regular expression?

Comment (bbk79) [reply plugin.video.iranintl/default.py]: Problem is, all I get is 3 separate 'return "https://...";' lines as per the above JS function. They'll be all equally 'valid'. I can make the assumption that the first one is the one I'm looking for which may or may not be true depending on the next change they make.

Comment (bbk79) [reply plugin.video.iranintl/default.py]: I guess that's arguably still a slight improvement over a hard-coded URL? Thoughts on using the first m3u8 URL I find in the page?

Comment (basrieter) [reply plugin.video.iranintl/default.py]: The pickOrigin now always returns "https://live.livetvstream.co.uk/LS-63503-3/index.m3u8" so I would take the first of the 3 matches

Comment (bbk79) [reply plugin.video.iranintl/default.py]: Done!

File: plugin.video.iranintl/default.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    html = response.read()
+
+    liveVideoUrls = re.findall('https://.*?.m3u8', html)
+
+    if len(liveVideoUrls) > 0:
+        xbmcplugin.addDirectoryItem(handle=addon_handle, url=liveVideoUrls[0], listitem=li)
+
+except urllib2.URLError as e:
+    if hasattr(e, 'reason'):
+        print 'We failed to reach a server. Reason: ', e.reason
```
Comment (basrieter) [inline plugin.video.iranintl/default.py]: `print ''` should be `print('')`

File: plugin.video.iranintl/default.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    elif hasattr(e, 'code'):
+        print 'The server couldn\'t fulfill the request. Error code: ', e.code
```
Comment (basrieter) [inline plugin.video.iranintl/default.py]: `print ''` should be `print('')`

**Pull request merged**